packages/core/src/llm-core/utils/similarity.ts

```
@@ -0,0 +1,141 @@
+        return dotProduct / (magnitude1 * magnitude2)
+    }
+
+    public static calculate(str1: string, str2: string): SimilarityResult {
+        if (!str1 || !str2) {
+            throw new Error('Input strings cannot be empty')
+        }
+
+        const text1 = TextTokenizer.normalize(str1)
+        const text2 = TextTokenizer.normalize(str2)
+
+        const cosine = this.cosineSimilarity(text1, text2)
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)